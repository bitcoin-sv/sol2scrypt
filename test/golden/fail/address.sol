contract C {

    uint immutable maxBalance;

    address private owner;

    constructor(address _reference) {
        owner =  _reference;
        maxBalance = _reference.balance;
    }


    function isBalanceTooHigh(address _other) public view returns (bool) {
        return _other.balance > maxBalance;
    }
    
    function transfer() external payable {
        payable(owner).transfer(msg.value);
    }


    function send(address payable _to) external payable {
        bool sent = _to.send(msg.value);
        require(sent, "Failed to send Ether");
    }

    function call(address payable _to) external payable {
        //(bool sent, bytes memory data) = _to.call{value: msg.value}("");
        //require(sent, "Failed to send Ether");
        _to.call("data");
    }
}