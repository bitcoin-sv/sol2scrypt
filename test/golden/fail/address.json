[
    {
        "src": "test/golden/fail/address.sol:9:22:9:40",
        "type": "Error",
        "message": "unsupported expression: `_reference.balance`"
    },
    {
        "src": "test/golden/fail/address.sol:14:16:14:30",
        "type": "Error",
        "message": "unsupported expression: `_other.balance`"
    },
    {
        "src": "test/golden/fail/address.sol:22:19:22:34",
        "type": "Error",
        "message": "unsupported type: `address payable`"
    },
    {
        "src": "test/golden/fail/address.sol:27:19:27:34",
        "type": "Error",
        "message": "unsupported type: `address payable`"
    }
]